### ISSUE TITLE
Add cover to sidecar files

### ISSUE THREAD

**Issue by captainkirk99** [SpatioTemporal/STAREmaster#38]
We wish to add a "cover" to the sidecar file.

The cover will express the coverage of the data in the file, and allow for easy matching against search requirements.

The cover is expressed either as:
1 - a 1-D array of STARE indicies.
2 - an array of STARE intervals.

**Comment by captainkirk99** [comment]
@NiklasPhabian @michaelleerilee where do I get the cover from?

Is there a STARE library function for it?

Is it different from the STARE indicies of the lowest-resolution data in the file?

**Comment by NiklasPhabian** [comment]
Do you want an index lookup for a region described by G-Rings? AFAI can see, this is not directly exposed in STARE yet. However, the underlying SpatialInterface is in fact based on G-Rings / 0-constraints / great circles. That means that you can get the cover by using STARE::ConvexHull by providing the 4 intersection points of the G-Rings. A bit backwards ...

**Comment by captainkirk99** [comment]
G-Ring - a series of points connected by geodesics.
The MODIS data files apparently have some G-Ring info.
Michael suggests using non-convex hull function.

**Comment by michaelleerilee** [comment]
I have added a cover calculation (including writing to the sidecar) on the branch cover1.

**Comment by michaelleerilee** [comment]
I've got the test working on my local branch -- I'll push soon.

Constructing a cover is taking 2-3 minutes for MOD05_L2 because I'm using all points on the perimeter of the data array. This is overkill for a number of reasons. MOD05_L2 has GRING (geodesic ring) information in the EOS metadata. See GROUP=GRING therein. I've parsed this in Python--but it would be nice to do the following.

- Check a granule for GRING information
- If it exists, parse the GRING and extract the points to use as a perimeter for the cover.
- If it doesn't, use our existing calculation, possibly striding through the data with a stride greater than 1 (current stride is 1).

I do not currently know how to get the GRING info from the metadata using C++.  Shouldn't be too hard though.
